cmd/nginx/main_test.go

```diff
@@ -18,46 +18,51 @@ package main
 
 import (
 	"fmt"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/ingress-nginx/internal/file"
+	"k8s.io/ingress-nginx/internal/ingress/controller"
 	"os"
 	"syscall"
 	"testing"
 	"time"
 
-	"k8s.io/ingress-nginx/internal/file"
-	"k8s.io/ingress-nginx/internal/ingress/controller"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func TestCreateApiserverClient(t *testing.T) {
-	home := os.Getenv("HOME")
-	kubeConfigFile := fmt.Sprintf("%v/.kube/config", home)
-
-	cli, err := createApiserverClient("", kubeConfigFile)
-	if err != nil {
-		t.Fatalf("Unexpected error creating Kubernetes REST client: %v", err)
-	}
-	if cli == nil {
-		t.Fatal("Expected a REST client but none returned.")
-	}
-
-	_, err = createApiserverClient("", "")
+	_, err := createApiserverClient("", "")
 	if err == nil {
-		t.Fatal("Expected an error creating REST client without an API server URL or kubeconfig file.")
+		t.Fatal("expected an error creating REST client without an API server URL or kubeconfig file.")
 	}
 }
 
 func TestHandleSigterm(t *testing.T) {
-	home := os.Getenv("HOME")
-	kubeConfigFile := fmt.Sprintf("%v/.kube/config", home)
+	clientSet := fake.NewSimpleClientset()
+
+	ns := "test"
+
+	cm := createConfigMap(clientSet, ns, t)
```

Comment:
Not sure to understand why this is here?

Reply:
Since `controller.NewNGINXController(conf, nil, fs)` make a call to the store. The store sets a configuration and requires a configmap. If it is not created then there will be a segmentation fault.